cmrtlib/linux/CMakeLists.txt

```diff
@@ -90,8 +91,26 @@ endif()
 set_target_properties( igfxcmrt PROPERTIES 
   FOLDER CM_RT 
   COMPILE_DEFINITIONS "${CMRT_DEFINES}" 
-  OUTPUT_NAME "igfxcmrt${TARGET_MODIFIER}"
+  OUTPUT_NAME "libigfxcmrt${TARGET_MODIFIER}"
   )
+
+# set the version, aligned with media driver
+if(NOT DEFINED MEDIA_VERSION OR MEDIA_VERSION STREQUAL "")
+    set(MEDIA_VERSION "1.0.0")
+endif()
+set(CMRT_VERSION ${MEDIA_VERSION})
+string(REPLACE "." ";" CMRT_VERSION_LIST ${CMRT_VERSION})
+list(GET CMRT_VERSION_LIST 0 CMRT_VERSION_MAJOR)
+
+set_target_properties(igfxcmrt PROPERTIES VERSION ${CMRT_VERSION})
+set_target_properties(igfxcmrt PROPERTIES SOVERSION ${CMRT_VERSION_MAJOR})
 set_target_properties( igfxcmrt PROPERTIES PREFIX "")
 target_link_libraries( igfxcmrt dl va rt ${GCC_SECURE_LINK_FLAGS})
 
+ADD_CUSTOM_TARGET(link_target ALL COMMAND ${CMAKE_COMMAND} -E create_symlink libigfxcmrt${TARGET_MODIFIER}.so igfxcmrt${TARGET_MODIFIER}.so)
+
+if(DEFINED CMAKE_INSTALL_LIBDIR AND (NOT CMAKE_INSTALL_LIBDIR STREQUAL ""))
+    install(TARGETS igfxcmrt LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT media NAMELINK_SKIP)
+    install(TARGETS igfxcmrt LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT media NAMELINK_ONLY)
```

Comment:
To be honest I do not favor self-packaging support in the repositories. I believe this is responsibility of the Linux distributions maintainers. I actually would like to see all the packaging code to be completely removed going forward. For these install commands this would mean remove code starting from "COMPONENT" keyword.

However, since packaging is currently here, you have wrong assumption that all targets should go to the same package. No, usually 2 packages are created:
1. runtime package which should hold libigfxcmrt.so.x.y.z and libigfxcmrt.so.x -> libigfxcmrt.so.x.y.z
2. development package which should hold libigfxcmrt.so and *.h and *.pc files

Please, decide how you will proceed with this....

Reply:
Avoid the "backwards" compatibility that you're trying to artificially create.  It will be a nightmare for distributions to know which ABI version they should use.